bundles/org.openhab.core.io.transport.serial/pom.xml

```diff
@@ -19,6 +19,12 @@
       <artifactId>org.openhab.core.io.console</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>commons-net</groupId>
+      <artifactId>commons-net</artifactId>
+      <version>3.7.2</version>
+      <scope>compile</scope>
+    </dependency>
```

Comment:
This will never work. You cannot manage dependencies of one bundle through another. The only way to make sure 3.7.2 is used by RXTX is by changing the Karaf feature files and making sure that only 3.7.2 of commons-net is installed. The RXTX bundle will use any available commons-net bundle that satisfies its specified version range. So it may even use 3.3 when it is installed eventhough 3.7.2 is available.

Reply:
Thanks for the explanation, I already feared that I need to touch the karaf stuff as well. Since I dont know much about karaf, I have it a try but obviously I failed :P

So do we have to wait for https://github.com/NeuronRobotics/nrjavaserial/issues/207 that I have just opened? Because I think think might take years again until we get a new release (from experiences in the past). Or is there any other option? I.e. that we leave nrjavaserial as it is, like compiling against 3.3, but at runtime only have 3.7.2 available and thus the bug is fixed for us?

It is really annoying to work with serial ports in openHAB at the moment. I would like to concentrate on working on features and improving stuff, rather than always getting obstructed by randomly working/not working thinghandlers because there are port available or not.

Reply:
> Because I think think might take years again until we get a new release (from experiences in the past).

Since the dependency change has already been merged now, I think there will be a release in time before OH 3.1.

> Or is there any other option?

If you update all core/add-on bundles/features to use commons-net 3.7+ you can guarantee that the current nrjavaserial version will also use the updated version. So that means you need to update this PR and probably create a PR for add-ons too.

Reply:
>I think there will be a release in time before OH 3.1.
That would be great. Let's hope for it :)

>If you update all core/add-on bundles/features to use commons-net 3.7+ you can guarantee that the current nrjavaserial version will also use the updated version. So that means you need to update this PR and probably create a PR for add-ons too.

For now I gave this a try, since we found out that we cannot remove the commons.net dependency due to nrjavaserial. So I adusted this PR and also created https://github.com/openhab/openhab-addons/pull/9921 to update the addons.

I guess it doesn't hurt to have it set to the latest version and if nrjavaserial bumps up its version to get compiled against commons-net-3.7.2, we can bump up nrjavaserial in addition.

Please let me know if I forgot anything.